cmd/juju/application/deployer/deployer.go

```diff
@@ -551,9 +547,14 @@ func (d *factory) newDeployBundle(_ charm.Schema, ds charm.BundleDataSource) dep
 }
 
 func (dk *localPreDeployerKind) CreateDeployer(_ context.Context, d factory) (Deployer, error) {
+	deployCharm := d.newDeployCharm()
+	if deployCharm.baseFlag.Empty() {
+		return nil, errors.Errorf("must provide a base with the `--base` flag when deploying pre-deployed local charms")
+	}
 	return &predeployedLocalCharm{
-		deployCharm:  d.newDeployCharm(),
+		deployCharm:  deployCharm,
 		userCharmURL: dk.userCharmURL,
+		base:         deployCharm.baseFlag,
```

Comment:
I guess the base flag the user passes in here is validated later (aside from checking if its empty), but is there any reason not to just validate it here and fail fast?

Reply:
The base here in `baseFlag` is already a parsed `base.Base`, so we already know that this is a valid base
https://github.com/juju/juju/blob/3.5/cmd/juju/application/deploy.go#L790-L809

It is hard todo much more validation client side, since this is a pre-deployed charm we don't have access to any information about the charm, other than it's charm URL.

There is a function, `validateCharmBaseWithName`, which can be moved here